### PR TITLE
Add a swagger definition for UI config settings and API Routes

### LINKED ISSUE #3777: API proposal for Logviewer UI config (#3766)

### PR DESCRIPTION
**Pull request by AlirieGray**
Closes #3777.

This is an API proposal for solving: https://github.com/influxdata/chronograf/issues/3713 and https://github.com/influxdata/chronograf/issues/3714. 

The user needs a way to configure their log viewer display settings and persist it on the backend per organization. 

This PR uses the existing config/:section endpoint and introduces a Swagger definition (`LogViewerConfig`) for a new section (`logViewer`). This config has two properties: a `columns` array and a `columnOptions` object with a `severity` property that defines the colors and names for the severity levels associated with logs.

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: server/swagger.json [hunk 1]
```diff
@@ -5030,27 +5136,90 @@
...
+        "name": {
+          "description": "Name is the user-facing name of the hex color",
+          "type": "string"
+        }
+      }
+    },
     "RenamableField": {
       "description":
-        "renamableField describes a field that can be renamed and made visible or invisible",
+        "Describes a field that can be renamed and made visible or invisible",
```
Comment (nhaugo) [inline server/swagger.json]: Would it be better to have a less general purpose type that is something like: LogColumn
```
{
  "tag": "tag-value", // influx 2 data storage meaning of tag, i.e. everything is a tag
  "display-text": "the displayed text"
  "widget": "dot-text",
  "pos": 0,
  "color": "some hex value"
} 
```

Comment (jaredscheib) [reply server/swagger.json]: that could be useful. `RenamableField` is a model already being used for table graph columns, so it seemed like something we could reuse here

File: server/swagger.json [hunk 2]
```diff
@@ -5030,27 +5136,90 @@
...
+      "properties": {
+        "id": {
+          "description": "ID is the unique id of the cell color",
+          "readOnly": true,
+          "type": "string"
+        },
+        "type": {
+          "description": "Type is how the color is used",
+          "type": "string",
+          "enum": ["emergency", "alert", "critical", "error", "warning", "notice", "info", "debug"]
```
[the remark this answers is not in the document]
Comment (jaredscheib) [reply server/swagger.json]: word. can a user edit these? or add new ones? or delete them? or could you express the use case for not being a fixed list? we did this based on provided info as well as the ui design here: https://github.com/influxdata/chronograf/pull/3753.

[the remark this answers is not in the document]
Comment (nhaugo) [reply server/swagger.json]: There are multiple RFCs we want to eventually support, there are deprecated forms of these in the RFC we followed, there is no standard setup for the telegraf logparser format and we will want to support custom log setups.

[the remark this answers is not in the document]
Comment (jaredscheib) [reply server/swagger.json]: Sounds good -- thanks for the context

**Comment by jaredscheib** [comment]
Post-discussion/feedback, we realized that the data model we'd collaborated on didn't seem to allow for some of the essential aspects of the requirements here. This makes me think that a more detailed specification of the UI requirements would be merited in an API proposal PR, rather than inferring them from the UI designs (in this case available here at #3753).

At any rate, specifically, we didn't see how what we'd come up with had a few issues that it didn't seem to address:
1) allow for a specific configuration of the severity colors themselves;
2) allow a single fixed value for the format of the icon/text/iconText to be displayed;
3) the position

This resulted from discussion with @ischolten who was implementing the client, where we realized that we were missing some data and structure that was necessary.

Additionally, after discussion with @alexpaxton and @ischolten, we agreed that a color's `name` and `hex` were not both necessary and having both was confusing since they are inextricably bound, so we went with just `Name`.

Finally, the explicit `position` also felt cumbersome, and we preferred to re-encode this implicitly using the order of the array structure, because we already have a convention and UI component for the drag-and-drop re-ordering that this would be subject to, and it works with re-ordering an array. To get it to work with a new `position` property, seemed unnecessary and confusing.

In light of all of this, and the earlier discussion with @goller, we simplified our API proposal and schema and added two commits to address them. We'd love another pass from @goller.

**Comment by jaredscheib** [comment]
Post-discussion/feedback, we realized that the data model we'd collaborated on didn't seem to allow for some of the essential aspects of the requirements here. This makes me think that a more detailed specification of the UI requirements would be merited in an API proposal PR, rather than inferring them from the UI designs (in this case available here at #3753).

At any rate, specifically, we didn't see how what we'd come up with had a few issues that it didn't seem to address:
1) allow for a specific configuration of the severity colors themselves;
2) allow a single fixed value for the format of the icon/text/iconText to be displayed;
3) the position

This resulted from discussion with @ischolten who was implementing the client, where we realized that we were missing some data and structure that was necessary.

Additionally, after discussion with @alexpaxton and @ischolten, we agreed that a color's `name` and `hex` were not both necessary and having both was confusing since they are inextricably bound, so we went with just `Name`.

Finally, the explicit `position` also felt cumbersome, and we preferred to re-encode this implicitly using the order of the array structure, because we already have a convention and UI component for the drag-and-drop re-ordering that this would be subject to, and it works with re-ordering an array. To get it to work with a new `position` property, seemed unnecessary and confusing.

In light of all of this, and the earlier discussion with @goller, we simplified our API proposal and schema and added two commits to address them. We'd love another pass from @goller.

File: server/swagger.json [hunk 3]
```diff
@@ -5143,6 +5143,60 @@
...
+              "type": {
+                "type": "string"
+              },
+              "value": {
+                "type": "string"
+              }
+            }
+          }
+        },
+        "mappings": {
```
Comment (AlirieGray) [inline server/swagger.json]: What does "mappings" mean in this context? For example, when a developer in the future wants to add another editable field to the columns, how do they know if what data should fall under "formatting" or "mappings"? @goller

Comment (goller) [reply server/swagger.json]: @AlirieGray I actual prefer only having one field, `formatting`.  This design comes from some desire to split those meanings.

If it is too difficult to determine which context it falls under (and I think it is too difficult) then that's a good hint it should change to just one.

File: server/swagger.json [hunk 4]
```diff
@@ -5143,6 +5143,60 @@
...
+          "description": "Unique identifier name of the column",
+          "type": "string"
+        },
+        "formatting": {
+          "description": "Composable formatting options for the column",
+          "type": "array",
+          "items": {
+            "description":"Type and value of a formatting option",
+            "type": "object",
+            "required": ["type", "value"],
```
[the remark this answers is not in the document]
Comment (goller) [reply server/swagger.json]: Not sure here... I don't really follow this JSON:

```json
          {
            "type": "text",
            "value": "dotText"
          }
```

I had imagined that the type would be "text" or "dot".  I'm not sure what the value is.

**Pull request merged**